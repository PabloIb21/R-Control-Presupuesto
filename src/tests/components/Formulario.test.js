import React from 'react';
import { mount } from 'enzyme';
import Formulario from '../../components/Formulario';

describe('<Formulario />', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = mount(
      <Formulario
        guardarGasto={() => {}}
        guardarCrearGasto={() => {}}
      />
    );
  });

  test('mount', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('should display error when the form is sent empty', () => {
    wrapper.find('form').simulate('submit');

    expect(wrapper.find('.error').exists()).toBe(true);
    expect(wrapper.find('.error').text()).toBe('Ambos campos son obligatorios o presupuesto incorrecto');
  });

  test('should display no error when the form is sent correctly', () => {
    wrapper.find('input').at(0).simulate('change', { target: { value: 'Transporte' } });
    wrapper.find('input').at(1).simulate('change', { target: { value: '300' } });
    wrapper.find('form').simulate('submit');

    expect(wrapper.find('.error').exists()).toBe(false);
  });
});