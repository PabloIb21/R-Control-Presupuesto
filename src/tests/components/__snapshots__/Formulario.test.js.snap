// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Formulario /> mount 1`] = `
<form
  onSubmit={[Function]}
>
  <h2>
    Agrega tus gastos aqu√≠
  </h2>
  <div
    className="campo"
  >
    <label>
      Nombre Gasto
    </label>
    <input
      className="u-full-width"
      onChange={[Function]}
      placeholder="Ej. Transporte"
      type="text"
      value=""
    />
  </div>
  <div
    className="campo"
  >
    <label>
      Cantidad Gasto
    </label>
    <input
      className="u-full-width"
      onChange={[Function]}
      placeholder="Ej. 300"
      type="number"
      value={0}
    />
  </div>
  <input
    className="button-primary u-full-width"
    type="submit"
    value="Agregar Gasto"
  />
</form>
`;
